### DISCUSSION TITLE
Min accuracy grace at the start of each test

### DISCUSSION THREAD

**Discussion by Otard95** [monkeytypegame/monkeytype#1492] [Ideas] [answered]
Hey!

I'm not the best typist, but i want to get better, which is why I started using monkeytype and crucially the 'min accuracy' setting.
I got it set to 95% as I don't want to pick up any bad habits. The issue is that when using this setting any mistake in the first ~20 characters will instantly fail you, which I find more annoying then helpful. However I'd still like to use the 'min accuracy' setting.

My suggested solution is a new setting, 'min accuracy grace'(the name is not important). The setting will modify the accuracy calculation, specifically in `src/js/test/test-timer.js:41`, to reduce the impact of errors, based on the `minAccCustom` config variable and the total number of keystrokes in the current test.

I did implement the feature in a [fork](https://github.com/Otard95/monkeytype), and wanted to make a PR, but was unable(presumably i don't have permission). However you are welcome to take a look at the implementation in the fork.

**Comment by unrealapex** [comment, marked as answer]
What branch did you use for the PR? If you created a new branch for the changes, you might need to explicitly selected it as the head of the PR.

**Reply by Otard95** [reply]
I did my work on master but is it not the base branch(the on you merge in to) that determines the permission needed? 

I could test another branch i guess.

**Reply by Otard95** [reply]
Yeah, you where indeed right! Thanks @UnrealApex!

**Comment by rpnfan** [comment]
Does not seem to be implemented in 2025, but indeed would be useful. It does not make sense to cut off a test, when you just had started a bit slow and very likely can catch up.

**Reply by Otard95** [reply]
I did create a [PR](https://github.com/monkeytypegame/monkeytype/pull/1494) for this, so might be the bug is back.
